bundler/lib/bundler/source_list.rb

```diff
@@ -221,7 +225,7 @@ def warn_on_git_protocol(source)
     end
 
     def equivalent_sources?(lock_sources, replacement_sources)
-      lock_sources.sort_by(&:identifier) == replacement_sources.sort_by(&:identifier)
+      lock_sources.sort_by(&:hash) == replacement_sources.sort_by(&:hash)
```

Comment:
sorting by hash is almost never stable

Reply:
Good catch, the idea still works though I think? We want to sort by something that's based on the keys used to compute this hash, but that is stable?

Reply:
Actually, all we care about here is that sorting is stable within the same invocation. And that's guaranteed, right?